### DISCUSSION TITLE
JoinColumn attribute should be mark as Attribute::IS_REPEATABLE ?

### DISCUSSION THREAD

**Discussion by mar20** [doctrine/doctrine2#9600] [General]
It JoinColumn attribute should be mark as Attribute::IS_REPEATABLE ? When you add e.g

#[ORM\JoinColumn(nullable: false)] - **wrong added line** 
#[ORM\JoinColumn(name: 'invoice_id', referencedColumnName: 'id', nullable: true)]

then then findOneBy(['invoice'=>$invoice]) - build wrong sql query with double invoice_id in where part

**Comment by derrabus** [comment]
I'm sorry, I don't understand the question.

**Reply by mar20** [reply]
`#[Attribute(Attribute::TARGET_PROPERTY | Attribute::IS_REPEATABLE) **<---I ask about this part]**  
final class JoinColumn implements Annotation`

**Reply by mar20** [reply]
Because when you add #[ORM\JoinColumn() twice working try use findOneBy it crash sql

**Reply by derrabus** [reply]
You need multiple join columns if the foreign entity has a composite primary key. So allowing this attribute to be repeatable looks correct to me.

**Reply by mar20** [reply]
ok, right. But may be better to use ORM\JoinColumns for this case ?

**Reply by derrabus** [reply]
No, `JoinColumns` is not an attribute.
